xbmc/addons/addoninfo/AddonInfo.cpp

```diff
@@ -283,12 +283,18 @@ std::vector<AddonInstanceId> CAddonInfo::GetKnownInstanceIds() const
   if (!m_supportsInstanceSettings)
     return singletonInstance;
 
+  std::vector<AddonInstanceId> ret;
+
+  if (!XFILE::CDirectory::Exists(StringUtils::Format("special://profile/addon_data/{}/", m_id)))
+  {
+    ret.emplace_back(ADDON_FIRST_INSTANCE_ID);
+    return ret;
+  }
+
   const std::string searchPath = StringUtils::Format("special://profile/addon_data/{}/", m_id);
   CFileItemList items;
   XFILE::CDirectory::GetDirectory(searchPath, items, ".xml", XFILE::DIR_FLAG_NO_FILE_DIRS);
```

Comment:
@emveepee sorry, I missed this PR, but would suggest to optimize this a bit to avoid two string instances with exactly the same content.

```c++
std::vector<AddonInstanceId> ret;
const std::string searchPath = StringUtils::Format("special://profile/addon_data/{}/", m_id);

if (!XFILE::CDirectory::Exists(searchPath))
{
  ret.emplace_back(ADDON_FIRST_INSTANCE_ID);
  return ret;
}

CFileItemList items;
XFILE::CDirectory::GetDirectory(searchPath, items, ".xml", XFILE::DIR_FLAG_NO_FILE_DIRS);
```

Reply:
@ksooo I don't even remember submitting the PR but I need to check what happens when special://masterprofile doesn't exist.

Reply:
@emveepee I will open the PR for the optimization I suggested. Not sure I got the message across and I'm very short on time currently. What you say about masterprofile might be valid, but is not all related to my suggestion.